wpilibj/src/main/java/edu/wpi/first/wpilibj/geometry/Translation2d.java

```diff
@@ -0,0 +1,164 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2019 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package edu.wpi.first.wpilibj.geometry;
+
+/**
+ * Represents a translation in 2d space.
+ * This object can be used to represent a point or a vector.
+ *
+ * <p>This assumes that you are using conventional mathematical axes.
+ * When the robot is placed on the origin, facing toward the X direction,
+ * moving forward increases the X, whereas moving to the left increases the Y.
+ */
+@SuppressWarnings({"ParameterName", "MemberName"})
+public class Translation2d {
+  private double m_x;
+  private double m_y;
```

Comment:
I think these should be final?

Reply:
Making these final means that I have to explicitly set them to zero in the 0-arg `Translation2d` ctor. Should I do that?

Reply:
Fixed in 9c88e6e